File: OpenCv-Cpp/RotationOperations/RotationOperations.cpp
#include <iostream>
#include <string>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>

int main()
{
	std::string path;
	std::cout << "Give Path for Image: ";
	std::cin >> path;
	// Load Image to a Matrix
	cv::Mat image;
	image = cv::imread(path);

	// make sure data is loaded
	if (!image.data)
	{
		std::cout << "\nImage File data not loaded.\nExiting...";
		return -1;
	}

	//Angle in degres
	double Angle = 45;

	//Image center
	cv::Point2f center = cv::Point(image.cols / 2, image.rows / 2);

	// Rotation Matrix
	cv::Mat rMatrix;

	//get Rotation Matrix
	rMatrix = cv::getRotationMatrix2D(center, Angle, 1);

	//Rotate with affine transform
	cv::Mat rotatedImage;
	cv::Size size = cv::Point(image.cols, image.rows);
	cv::warpAffine(image, rotatedImage, rMatrix, size);

	// Display Matrix(image) in GUI.
	cv::imshow("Original Image", image);
	cv::imshow("Rotated Image", rotatedImage);

	// wait with delay for any key from GUI.
	cv::waitKey(0);
	cv::destroyAllWindows();
	return 0;
}

File: OpenCv-Python/RotationOperations/RotationOperations.py
import cv2

path = input("Enter path for Image:")

# Read image matrix from Image File
image = cv2.imread(path)

# check if image is read
if image is None:
    print('Image File data not loaded.\nExiting...')
    exit()

# Angle in degres
Angle = 45

# Image center
center = (image.shape[1]/2, image.shape[0]/2)

# get Rotation Matrix
RMatrix = cv2.getRotationMatrix2D(
    center=center, angle=Angle, scale=1)

# Rotate with affine transform
Rotated_Image = cv2.warpAffine(src=image, M=RMatrix,
                               dsize=(image.shape[1], image.shape[0]))

# show the images in a GUI
cv2.imshow('original', image)
cv2.imshow('Rotated Image', Rotated_Image)

# after delay wait for any key from GUI
cv2.waitKey(0)

# destroy or free any remaining GUI functionality
cv2.destroyAllWindows()
